### ISSUE TITLE
Drop support for missing MIR

### ISSUE THREAD

**Issue by oli-obk** [solson/miri#211]
Stuff to do

- [x] use `exchange_malloc` in the `box` unop
- [ ] remove the hacks that catch some common missing MIR functions

**Comment by eddyb** [comment]
cc @RalfJung

**Comment by RalfJung** [comment]
Sounds fair enough. I am not sure about the first point -- I actually feel like miri really should have some more sophisticated logging infrastructure, so that I can selectively turn on/off tracing for different modules or so. And some things are pretty hard to see despite all the verbosity of a full trace, like which arguments are passed on a function call.

I suppose this is blocked by #202?

**Comment by oli-obk** [comment]
You can already do the module-wise logging. We use `env_logger`

**Comment by RalfJung** [comment]
Okay so it's mostly a matter of documentation then. And maybe of organizing things do that the outputs of individual modules make more sense. ;)  Like, I may want to say "trace instructions that are executed", which should exactly print every stmt and terminator that is executed, and nothing else. But I don't want to see details like symbol resolution or vtable caches.

**Comment by oli-obk** [comment]
We can remove the blocker on #202 by

a) adding appveyor and a travis mac build, or
b) enabling MIR generation unconditionally for the standard library

Though a) would remove the feature that miri can easily interpret stuff for other targets.

**Comment by oli-obk** [comment]
One checkbox solved by #351

**Comment by RalfJung** [comment]
I think we are basically ready to do this -- except that #202 is still open. @oli-obk do you think that should block this? Can we at least get a Travis runner with a 32bit platform to make sure those are covered for a unix system as well?

Also you added "add some flag to not report the libstd boilerplate in MIRI_LOG" to the list. That'd be nice to have but I'd prefer to not block on this, and open an issue instead.

**Comment by oli-obk** [comment]
I moved it out to a separate issue.
